docs/UserGuide.md

```diff
@@ -10,21 +10,39 @@ title: User Guide
 ## GradPad
 
-GradPad is a one-stop solution to module management for Computer Science Undergraduates.
+We are a team of **Computer Science Undergraduates** working together to solve **module management** and **graduation planning** problems
+faced by fellow Computer Science students. Below is the detailed User Guide for effective use of GradPad which can help you
+ease and improve your **module planning process**.
+
+GradPad is a one-stop solution to **module management** for Computer Science Undergraduates.
 Planning for modules has always been a tedious process but it does not have to be.
 
-The current approach to planning and tracking graduation requirements is to open up tabs after tabs of NUS resources which can
-be messy at times. So, our team has come up with the idea of an easy-to-use, all-in-one application that
-can ease the process of module management for Computer Science Undergraduates.
+The current approach to planning and tracking graduation requirements is to open up tabs after tabs of NUS resources such as 
+[NUSMods](https://nusmods.com/) and [websites](https://www.comp.nus.edu.sg/programmes/ug/cs/curr/) of Faculty of Computing, 
+which can be messy at times. 
+So, our team has come up with the idea of an **easy-to-use**, **all-in-one application** that
+can ease the process of **module management** for Computer Science Undergraduates.
 
-GradPad consolidates the modules you have taken and displays the remaining required modules to
-ease your module planning process. A module searching platform is also included in GradPad, providing easy
-navigation of modules.
+GradPad **consolidates** the modules you have taken and **displays** the remaining required modules to
+ease your module planning process. A **module searching platform** is also included in GradPad, providing easy
+**navigation** of modules.
 
 This user guide introduces what you can do with GradPad and the steps needed to do them. In addition,
-it includes a quick start guide that walks you through the installation and set-up process.
+it includes a [quick start guide](#quick-start) that walks you through the installation and set-up process.
 With this guide, you'll be ready to use GradPad in no time.
 
+--------------------------------------------------------------------------------------------------------------------
+## NUSMods integration  
+
+![NUSMods](images/nusmods.png)
+
+To streamline your module searching process with added convenience, GradPad integrates data from [NUSMods](https://nusmods.com/). With this, 
+you no longer have to visit NUSMods separately to look up module information. Instead, GradPad offers you all that data 
+in the same app you use to plan your modules itself.
+
+We have requested and was granted permission from NUSMods to utilise their platform to retrieve data such as NUS course catalogue
+and **module information**. These data are retrieved on the go in **realtime**, using the data provided by NUSMods.
```

Comment:
Sentence structure nit: These data are retrieved on the go in **realtime**, directly from NUSMods.